app/controllers/errors_controller.rb

```diff
@@ -13,7 +13,11 @@ def display_skip_to_main_navigation?
   private
 
   def fetch_exception
-    @exception = env['action_dispatch.exception']
+    @exception = if env['action_dispatch.request.path_parameters'][:status_code] == '404'
+       ActionController::RoutingError.new(:page_not_found)
```

Comment:
Why the controller fetch an exception to raise another?

Reply:
@tomas-stefano for some reason `env['action_dispatch.exception']` or `request. env['action_dispatch.exception']` always returns a `nil` value. This means by default every error is forced to render as an _internal server error_ issue even when it's not a `500` error. So one thing is to catch the error early before it's coerced by default to be a `500` error even when it isn't.

Do you see what I mean. please?

Reply:
So is the message "This means by default every error is forced to render as an internal server error issue even when it's not a 500 error"  sounds that is fixing a bug then?

Reply:
It is a bug. There was some work done to remove the Gaffe gem() about two years ago. This work may have accidentally introduced the bug. 

Here is a bug-fix by @igelstorm on this very issue from about a month ago (https://github.com/moneyadviceservice/frontend/pull/1789).

Reply:
I didn't understand. Is this PR fixing a bug or refactor existing feature?

Reply:
Looks like its refactoring an existing feature as well as addressing tech debt